Packs/OracleCloudInfrastructure/Integrations/OracleCloudInfrastructureEventCollector/OracleCloudInfrastructureEventCollector.py

```diff
@@ -0,0 +1,396 @@
+    - This function uses pagination, meaning it will make multiple request as needed to reach the desired amount of events.
+
+    Args:
+        client (Client): Client object for requests.
+        first_fetch_time (datetime): The start time to fetch events from.
+        max_fetch (int): The limit of events to fetch.
+
+    Raises:
+        DemistoException: If an error occurred while fetching events.
+
+    Returns:
+        tuple[list[dict[str, Any]], str]: A tuple of the events list and the last event time for next fetch cycle.
+    """
+    try:
+        response = audit_log_api_request(client=client, start_time=first_fetch_time.strftime(DATE_FORMAT))
+        events = json.loads(response.content)
+
+        if not events:
+            return [], first_fetch_time.strftime(DATE_FORMAT)
+
+        if isinstance(events, dict):
+            events = [events]
+
+        # pagination handling
+        while len(events) < max_fetch and (next_page := response.headers._store.get('opc-next-page')):  # type: ignore
+            current_start_time = add_millisecond_to_timestamp(events[-1].get('eventTime'))
+            response = audit_log_api_request(client=client, start_time=current_start_time, next_page=next_page[1])
+            events.extend(json.loads(response.content))
+
+        last_event_time = get_last_event_time(events, first_fetch_time)
+        events = add_time_key_to_events(events)
+
+    except Exception as e:
+        if events:
+            last_event_time = get_last_event_time(events, first_fetch_time)
+            events = add_time_key_to_events(events)
+            handle_fetched_events(events, last_event_time)
+            raise DemistoException(f'Error while fetching events: {e}') from e
+
+    demisto.info(f'OCI: {len(events)} Events fetched from start time: {first_fetch_time}.')
+    return events, last_event_time
+
+
+def handle_fetched_events(events: list[dict[str, Any]], last_event_time: str):
+    """Handles fetched events.
+    - Sends the events to XSIAM.
+    - Sets the last run for next fetch cycle.
+
+    Args:
+        events (list[dict[str, Any]]): Fetched events.
+        last_event_time (str): Last event time.
+    """
+    send_events_to_xsiam(events, vendor=VENDOR, product=PRODUCT)
+    demisto.info(f'OCI: {len(events)} events were sent to XSIAM at {datetime.now()}.')
+    if events:
+        demisto.setLastRun({"lastRun": last_event_time})
+        demisto.info(f'OCI: Set last run to {last_event_time}')
+    else:
+        demisto.info('OCI: No new events fetched, Last run was not updated.')
+
+
+''' Test module '''
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication.
+
+    Args:
+        client (Client): Client for SDK interaction and api requests.
+
+    Raises:
+        DemistoException: If an error occurred while testing.
+
+    Returns:
+        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+
```

Comment:
We might need to test the fetch here - we usually do it in integrations including the fetch command. 
Not sure about event collectors specifically though.